service/middleware.go

```diff
@@ -22,6 +25,9 @@ func (rm HTTPTestRequestMiddleware) GetMiddlewareFunction() func(http.Handler) h
 			r.URL.Scheme = ""
 			r.URL.Host = ""
 
+			requestURI, _ := strings.CutPrefix(r.RequestURI, "http://foo")
+			r.RequestURI = requestURI
+
```

Comment:
sorry to see this late, but I'd rather make this a config option (to be set by the test framework, or empty by default), perhaps:
```suggestion
if cfg.TestHostPrefix != "" {
			requestURI, _ := strings.CutPrefix(r.RequestURI, cfg.TestHostPrefix)
			r.RequestURI = requestURI
}

```